### PR TITLE
Migrate node from commonJS to ESM

### LINKED ISSUE #74: Transition from CJS to ESM
Currently, the user interface is built using CommonJS with Vite. When the build is running, the following warning is provided:

```
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
```

Transitioning the entire process to ESM from CJS will resolve deprecations and improve overall build.

### PR DESCRIPTION
**Pull request by flamingquaks**
This changes the project to use ESM rather than CommonJS as the target JS.

This resolves #74, including the vite CommonJS deprecation warnings. 

Updated `cdk.json` to use `tsx` rather than `ts-node` due to issues with `ts-node` with `esm`.

`NodeJsFunction` has a known issue resolving the default file. The workaround, added to each lambda function `entry` property can be found here: https://github.com/aws/aws-cdk/issues/21630#issuecomment-1299955735

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by peteconraws — DISMISSED** [review]
All Good

**Comment by flamingquaks** [comment]
@peteconraws - Can you re-approve this? After merging other PRs, this one had some changes required. All changes have been deployed & tested for regression.

**Review by peteconraws — APPROVED** [review]
All Good

**Pull request merged**
